docs/DeveloperGuide.md

```diff
@@ -567,64 +567,211 @@ testers are expected to do more *exploratory* testing.
+balance is shown.
+
+### Adding a transaction
+
+Prerequisite: There is only 1 person stored in Spend N Split. Their name is "Alex Yeoh".
+
+`addTransaction d=bread n=Alex Yeoh c=25.00 n=Self w=1 n=Alex Yeoh w=1`
+
+Expected Output in the Transaction List Panel: New transaction is added into the list.
+
+Expected Output in the Command Output Panel: New transaction added message is shown along with the transaction's details.
+
+
+### Deleting a Transaction
+
+Prerequisite: There is at least 1 transaction in the Transaction List Panel.
+
+`deleteTransaction 1`
+
+Expected Output in the Transaction List Panel: The first transaction in the list is deleted.
+
+Expected Output in the Command Output Panel: Delete transaction message is shown along with the transaction's details.
+
+`deleteTransaction 0`
+
+Expected Output in the Command Output Panel: Error message stating that there is an invalid command format.
+### Editing a Transaction's details
+
+Prerequisite: There is at least 1 transaction in the Transaction List Panel.
+
+`editTransaction 1 c=20`
+
+Expected Output in the Transaction List Panel: The first transaction in the list has their cost changed to 20.00, and retains
+the rest of their details.
+
+Expected Output in the Command Output Panel: Edit transaction message is shown along with the transaction's details.
+
+`editTransaction 0`
+
+Expected Output in the Command Output Panel: Error message stating that there is an invalid command format.
+
+### Listing multiple transactions with matching payers and payees.
+
+Prerequisite: There is only 1 transaction stored in Spend N Split. The payer is "Roy".
+
+`listTransaction Roy`
+
+Expected Output in the Transaction List Panel: The transaction is displayed.
+
+Expected Output in the Command Output Panel: List transaction message is shown and states "1 transaction(s) listed".
+
+`listTransaction Tan`
+
+Expected Output in the Transaction List Panel: Nothing is displayed.
+
+Expected Output in the Command Output Panel: List transaction message is shown and states "0 transaction(s) listed".
+
+### Duplicating a Transaction
+
+Prerequisite: There is at least 1 transaction in the Transaction List Panel.
+
+`duplicateTransaction 1 c=20`
+
+Expected Output in the Transaction List Panel: A new transaction is created with identical details to the transaction
+ at index 1, with the only difference being that the new transaction's cost is 20.00.
+
+Expected Output in the Command Output Panel: Duplicate transaction message is shown along with the transaction's details.
+
+`duplicateTransaction 0`
+
+Expected Output in the Command Output Panel: Error message stating that there is an invalid command format.
+
+### Updating a Portion of a Transaction
+
+Prerequisite: There is at least 1 transaction in the Transaction List Panel.
+
+`updatePortion 1 n=David w=0.5`
+
+Expected Output in the Transaction List Panel: A new payer (David) is added to the transaction at index 1. David's 
```

Comment:
Based on this expected, we should specify in the pre-requisite that David was not part of the transaction initially

Reply:
fixed